apps/passport-server/src/services/multiProcessService.ts

```diff
@@ -78,15 +130,23 @@ export class MultiProcessService {
     // to the worker farm in a promise.
     return traced(LOG_NAME, "verifySignaturePCD", async () => {
       logger("[MULTIPROCESS] verifying a semaphore signature");
-      return new Promise<boolean>((resolve, reject) => {
-        this.workers(JSON.stringify(pcd), (err: Error, result: boolean) => {
+      const payload = JSON.stringify(pcd);
+      // Are we already processing an identical job?
+      // Using the stringified payload as the cache key
+      if (this.jobs.verifySignaturePCD.has(payload)) {
```

Comment:
it feels like it should cache a result, i.e. a boolean, not a promise.

Reply:
But this would not work for simultaneous requests, because we can't cache a result until one of them has completed. So the first arrives, we look in the cache, nothing there, start a new worker. The second arrives, we look in the cache, nothing there, start a new worker, and so on. So long as all of the requests arrive before the first one resolves, then none of them use the cache, and this is how I would imagine it would work when receiving multiple feed poll requests at the same time.

Reply:
got it - makes sense!